### PR TITLE
TE5-T637-Practical-03

### PR DESCRIPTION
**Pull request by JeetMakadiya**

### CODE REVIEW

**Comment by JeetMakadiya** [comment]
Modal and Form is Created.
Screenshot from 2022-03-30 16-29-51

**Comment by JeetMakadiya** [comment]
Form Validations Added. Edit Button Added.
Screenshot from 2022-04-01 15-57-37

File: practical-03/src/components/CarForm.vue [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        carDetails: this.formData.formData.carDetails || "",
+        carPrice: this.formData.formData.carPrice || "",
+        carImgURL: this.formData.formData.carImgURL || "",
+      },
+      formModalId: this.modalId,
+    };
+  },
+  methods: {
+    onSubmit() {
+      console.log(this.car);
```
Comment (rajbhojani88) [inline practical-03/src/components/CarForm.vue]: console never have to be uploaded on git

File: practical-03/src/components/FormModal.vue [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    resetModal() {
+      this.name = "";
+      this.nameState = null;
+    },
+    handleOk(bvModalEvt) {
+      // Prevent modal from closing
+      bvModalEvt.preventDefault();
+      // Trigger submit handler
+      this.handleSubmit();
+    },
```
Comment (rajbhojani88) [inline practical-03/src/components/FormModal.vue]: is it used ?

File: practical-03/src/components/FormModal.vue [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      if (!this.checkFormValidity()) {
+        return;
+      }
+      // Push the name to submitted names
+      this.submittedNames.push(this.name);
+      // Hide the modal manually
+      this.$nextTick(() => {
+        this.$bvModal.hide("modal-prevent-closing");
+      });
+    },
```
Comment (rajbhojani88) [inline practical-03/src/components/FormModal.vue]: is it used ?

File: practical-03/src/components/FormModal.vue (hunk 2 above)
Comment (JeetMakadiya) [reply practical-03/src/components/FormModal.vue]: No it is not used. I have removed that.

File: practical-03/src/components/FormModal.vue (hunk 3 above)
Comment (JeetMakadiya) [reply practical-03/src/components/FormModal.vue]: No it is not used. I have removed that.

File: practical-03/dist/index.html [hunk 4]
```diff
@@ -0,0 +1 @@
+<!doctype html><html lang=""><head><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><link rel="icon" href="/favicon.ico"><title>practical-03</title><script defer="defer" type="module" src="/js/chunk-vendors.07ce5e60.js"></script><script defer="defer" type="module" src="/js/app.501b9eb6.js"></script><link href="/css/chunk-vendors.dcf11e75.css" rel="stylesheet"><link href="/css/app.45f6adcd.css" rel="stylesheet"><script defer="defer" src="/js/chunk-vendors-legacy.0cc29520.js" nomodule></script><script defer="defer" src="/js/app-legacy.4afc9fe4.js" nomodule></script></head><body><noscript><strong>We're sorry but practical-03 doesn't work properly without JavaScript enabled. Please enable it to continue.</strong></noscript><div id="app"></div></body></html>
```
Comment (rajbhojani88) [inline practical-03/dist/index.html:1]: blank line

File: practical-03/src/components/FormModal.vue [hunk 5]
```diff
@@ -0,0 +1,43 @@
+<template>
+  <div>
+    <b-modal
+      id="modal-prevent-closing"
+      ref="modal"
+      title="Add Car Details"
+      @show="resetModal"
```
Comment (rajbhojani88) [inline practical-03/src/components/FormModal.vue]: resetModal ?

**Pull request merged**